src/python/pants/backend/python/interpreter_cache.py

```diff
@@ -127,11 +144,14 @@ def setup(self, paths=(), filters=(b'',)):
     :returns: A list of cached interpreters
     :rtype: list of :class:`pex.interpreter.PythonInterpreter`
     """
+    pex_python_path_interpreter_paths = self.pex_python_path_list
     # We filter the interpreter cache itself (and not just the interpreters we pull from it)
     # because setting up some python versions (e.g., 3<=python<3.3) crashes, and this gives us
     # an escape hatch.
-    filters = filters if any(filters) else self._python_setup.interpreter_constraints
+    if not any(filters) and not pex_python_path_interpreter_paths:
```

Comment:
what if filters set contains conflicting filter requirements?  This can possibly be true when running `pants test ::` while having target level constraints.